### ISSUE TITLE
Error: Parsing Error when using {| |}

### ISSUE THREAD

**Issue by joris974** [gajus/eslint-plugin-flowtype#113]
Not sure if this should be reported here.
Flow 0.32 added the new syntax for exact object types:
```
type A = {| a: number |}
```
This syntax throws a parsing error on ES Lint.
```
Parsing error: Unexpected token
```
eslint v3.2.2
eslint-plugin-flowtype v2.17.1
flow-bin v0.32.0

**Comment by danharper** [comment]
Babel doesn't _yet_ support Flow's exact object type syntax.

https://github.com/babel/babylon/pull/104

So hold off on using them until the next Babel release :)

**Comment by chrmcg** [comment]
Could this issue be reopened? That PR was merged into [babel@6.10.0](https://github.com/babel/babylon/releases/tag/v6.10.0).

**Comment by danharper** [comment]
Why? Are you having issues with it? Exact object syntax should work fine.

**Comment by chrmcg** [comment]
Yes, I'm having the same issue as OP.
eslint v3.5.0
babel-eslint v7.1.1
eslint-plugin-flowtype v2.29.1
flow-bin v0.35.0

I thought it might be an issue with my eslint config, but even in a clean directory with an `.eslintrc.js` copied directly from the README and a single `test.js` containing 
```
// @flow
type A = {| a: number |}
```
`eslint .` still returns `  2:11  error  Parsing error: Unexpected token` for `test.js`.

**Comment by danharper** [comment]
What version of Babel do you have installed? You should be able to upgrade/re-install and get the latest Babel.

**Comment by danharper** [comment]
*Babylon, not Babel

**Comment by chrmcg** [comment]
Aha! Thanks for your help. My issue was that I had installed both Babel and eslint globally at some point and so I was depending on Babylon 6.8.4 without realizing it.

For the record, in my earlier comment I completely missed that it was Babylon 6.10 that introduced exact object syntax, not Babel 6.10.

Fixed by running `eslint .` as an npm script instead of directly from the command line. Thanks again.
